### ISSUE TITLE
[Bug] Impossible to override globale retention policy in a room with a bigger value.

### ISSUE THREAD

**Issue by finkyseb** [RocketChat/Rocket.Chat#20186]
### Description:

It seems impossible to specify a bigger room retention policy in the web interface.
The new value is written in the mongo database, but in the preferences of the room the the policy is set to the globale value.

### Steps to reproduce:

1. You must set a global retention policies ex: 30 days.
2. Go on a channel and try to set a value bigger than the global one You can do the same with the APi or directly in BDD (it's the same result)
3. The value is correctly written in the database
4. You can check that the value is always the general value not yours.
5. If you set a value lower than the global value it works and the new value is yours.

### Expected behavior:

We expect that if a value is specified on a room, this value override the globale retention value and that in all cases.
We don't want the application takes only the smaller value.

### Actual behavior:

Screenshots can be done, but I think the description is enough to reproduce the bug.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.9.1 to 3.10.3
- CENTOS (on docker): 
- Deployment Method: docker (swarm)
- Number of Running Instances: 12
- DB Replicaset Oplog: Enabled
- NodeJS Version:  12.18.4
- MongoDB Version: 4.2.11

### Client Setup Information

- Desktop App or Browser Version: insignificant
- Operating System: insignificant

**Comment by leManu** [comment]
Maybe the problem sits in this line ? https://github.com/RocketChat/Rocket.Chat/blob/119028b17f71df8797a46234cfee7cc6c8ee9217/app/retention-policy/server/cronPruneMessages.js#L50

maxAge is always 30...

**Comment by finkyseb** [comment]
I don't think so.
The value is set to 30 only if the value in room.retention is not set.
But it can't happened because we search only rooms with this attributes :
```
'retention.enabled': { $eq: true },
'retention.overrideGlobal': { $eq: true },
'retention.maxAge': { $gte: 0 },```

**Comment by finkyseb** [comment]
After more tests, it seems the room configuration is taken by the cron job correctly, but only the value in the webinterface in the room information is incorrect.
Anybody can confirm that ?

**Comment by EnCz** [comment]
> After more tests, it seems the room configuration is taken by the cron job correctly, but only the value in the webinterface in the room information is incorrect.
> Anybody can confirm that ?

Yes that seems to be the case. Retention Policy Settings are a little bit wonky at the moment. 

In Administration we have activated the retention policy but didn't apply it to any room:
![image](https://user-images.githubusercontent.com/7642122/104455264-56420a00-55a7-11eb-8ba4-1861fcd14d31.png)

So our users can overwrite this setting on a per-room basis. Still every room has this message in it's room info:
![image](https://user-images.githubusercontent.com/7642122/104455415-88ec0280-55a7-11eb-9bf0-8e3b0a1c11c9.png)

even though there is no retention policy set for that specific room.
 

If you go ahead and set a per-room policy it seems to work as intended (even though the box will still display "30" even if you e.g. define 5).
**Warning: ** Sometimes you need to save a per-room configuration twice.


We also have issues with threads. Those doesn't seem to get deleted by the retention policy even though we have this settings in administration:
![image](https://user-images.githubusercontent.com/7642122/104455693-f435d480-55a7-11eb-862a-cfebd2f35f6e.png)


So yes.. you can get it to work but retention policies seem to have a lot of little issues at the moment. I would be very happy if this gets resolved sometimes.

I also alredy opened two tickets right here:
https://github.com/RocketChat/Rocket.Chat/issues/19730
https://github.com/RocketChat/Rocket.Chat/issues/19731

Also someone on french posted here:
https://github.com/EsupPortail/esup-mod_rocketchat/issues/21



@sampaiodiego maybe you can say something about this ?

**Comment by finkyseb** [comment]
probably fix in #20349
